packages/kosu-genesis-cli/src/functions.ts

```diff
@@ -0,0 +1,294 @@
+        }
+    }
+    return validators;
+}
+
+/**
+ * Generate a "snapshot" of the PosterRegistry contract state (all posters) by
+ * re-playing `PosterRegistryUpdate` events from the Kosu EventEmitter contract.
+ *
+ * The resulting array contains the address and balance of each account at the
+ * specified snapshot block.
+ *
+ * @param kosu An initialized kosu.js instance.
+ * @param snapshotBlock The Ethereum block at which to stop replaying past event logs.
+ * @returns Promise resolving to snapshot poster info (see type definition).
+ */
+export async function snapshotPostersAtBlock(kosu: Kosu, snapshotBlock: number): Promise<SnapshotPoster[]> {
+    const posters: SnapshotPoster[] = [];
+    const postersMap: { [address: string]: SnapshotPoster } = {};
+
+    // get all events from deploy to snapshot block
+    const allLogs = await kosu.eventEmitter.getPastDecodedLogs({
+        fromBlock: 0,
+        toBlock: snapshotBlock,
+    });
+
+    for (const log of allLogs) {
+        const { decodedArgs } = log;
+        switch (decodedArgs.eventType) {
+            case "PosterRegistryUpdate": {
+                const { poster: address, stake: balance } = decodedArgs;
+
+                if (balance === "0") {
+                    delete posters[address];
+                }
+
+                if (!postersMap[address]) {
+                    (postersMap[address] as Partial<SnapshotPoster>) = { address };
+                }
+                postersMap[address].balance = balance;
+                break;
+            }
+            default:
+                continue;
+        }
+    }
+
+    for (const posterAddress of Object.keys(postersMap)) {
+        posters.push(postersMap[posterAddress]);
+    }
+
+    return posters;
+}
+
+/**
+ * Convert a Tendermint public key to a Tendermint address (also called node ID).
+ *
+ * @param publicKey Tendermint 32-byte public key buffer.
+ * @returns The corresponding Tendermint address string.
+ */
+export function publicKeyToAddress(publicKey: Buffer): string {
+    const digest = createHash("sha256");
+    const hash = digest.update(publicKey).digest();
+    const address = hash
+        .slice(0, 20)
+        .toString("hex")
+        .toUpperCase();
+    return address;
+}
+
+/**
+ * Convert the validator snapshot data to the JSON format expected by Tendermint.
+ *
+ * @param validators Raw validator snapshot data.
+ * @returns Tendermint-style genesis validator JSON.
+ */
+export function getTendermintValidators(validators: SnapshotValidator[]): GenesisValidator[] {
+    const genesisValidators: GenesisValidator[] = [];
+    for (const validator of validators) {
+        genesisValidators.push({
+            address: publicKeyToAddress(validator.publicKey),
+            pub_key: {
+                type: "tendermint/PubKeyEd25519",
+                value: validator.publicKey.toString("base64"),
+            },
+
+            // Note: power is calculated and set by the network
+            power: "0",
```

Comment:
If we don't specify power, the client will assume it's zero, is there any reason to explicitly use `0`?

Reply:
Nope -- I wasn't aware we could omit it. I will do that.